qiskit_experiments/test/mock_iq_backend.py

```diff
@@ -154,21 +154,32 @@ def __init__(
         ideal_beta=2.0,
         gate_name: str = "Rp",
         rng_seed: int = 0,
+        max_prob: float = 1.0,
+        offset_prob: float = 0.0,
     ):
         """Initialize the rabi backend."""
         self._error = error
         self._gate_name = gate_name
         self.ideal_beta = ideal_beta
 
+        if max_prob + offset_prob > 1:
+            raise ValueError("Probabilities need to be between 0 and 1.")
+
+        self._max_prob = max_prob
+        self._offset_prob = offset_prob
+
         super().__init__(iq_cluster_centers, iq_cluster_width, rng_seed=rng_seed)
 
     def _compute_probability(self, circuit: QuantumCircuit) -> float:
         """Returns the probability based on the beta, number of gates, and leakage."""
-        n_gates = sum(circuit.count_ops().values())
+        n_gates = circuit.count_ops()[self._gate_name]
 
         beta = next(iter(circuit.calibrations[self._gate_name].keys()))[1][0]
 
-        return np.sin(n_gates * self._error * (beta - self.ideal_beta)) ** 2
+        prob = np.sin(n_gates * self._error * (beta - self.ideal_beta)) ** 2
```

Comment:
What do you think about scaling error by 2 pi and calling it freq?

Reply:
Sounds good. See here: [ebae5a4](https://github.com/Qiskit/qiskit-experiments/pull/732/commits/ebae5a4fc76a8ebc380f73796edfe60612040877)